pkg/check/act/act.go

```diff
@@ -0,0 +1,209 @@
+		return fmt.Errorf("node %s: %w", upNodeName, uErr)
+	}
+	c.logger.Info("ACT file uploaded")
+
+	time.Sleep(3 * time.Second)
+
+	act := true
+	history := file.HistroryAddress()
+	timestamp := uint64(2)
+
+	// download act file
+	size, hash, err := upClient.DownloadActFile(ctx, file.Address(), &api.DownloadOptions{Act: &act, ActPublicKey: &publisher, ActHistoryAddress: &history, ActTimestamp: &timestamp})
+	if err != nil {
+		return fmt.Errorf("node %s: %w", upNodeName, err)
+	}
+
+	if !bytes.Equal(file.Hash(), hash) {
+		c.logger.Infof("Node %s. ACT file hash not equal. Uploaded size: %d Downloaded size: %d  File: %s", upNodeName, file.Size(), size, file.Address().String())
+		return errors.New("ACT file retrieval - hash error")
+	}
+
+	c.logger.Info("ACT file downloaded")
+
+	// download act file with wrong public key
+	notPublisher, _ := swarm.ParseHexAddress("010101010101010101010101010101010101010101010101010101010101010101")
+	_, _, notPErr := upClient.DownloadActFile(ctx, file.Address(), &api.DownloadOptions{Act: &act, ActPublicKey: &notPublisher, ActHistoryAddress: &history, ActTimestamp: &timestamp})
+	if notPErr == nil {
+		return fmt.Errorf("node %s: File downloaded with wrong public key successfully - this is an error", upNodeName)
+	}
+
+	c.logger.Info("ACT Access denied with incorrect public key")
+
+	// add grantees list
+	gFile := bee.NewBufferFile("grantees.json", bytes.NewBuffer([]byte(`{ "grantees": [
+        "02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e8",
+        "02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e9",
+        "02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12ee"
+    	]
+	}`)))
+
+	err = upClient.AddActGrantees(ctx, &gFile, api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return fmt.Errorf("node %s: add grantees error: %w", upNodeName, err)
+	}
+
+	c.logger.Info("ACT grantees added")
+
+	time.Sleep(3 * time.Second)
+
+	// list grantees
+	addresses, gErr := upClient.GetActGrantees(ctx, gFile.Address())
+	if gErr != nil {
+		return fmt.Errorf("node %s: GetActGrantees: %w", upNodeName, gErr)
+	}
+	if addresses == nil {
+		return fmt.Errorf("node %s: GetActGrantees: addresses is nil", upNodeName)
+	}
+	if len(addresses) != 3 {
+		return fmt.Errorf("node %s: GetActGrantees: addresses length is not 3", upNodeName)
+	}
+
+	c.logger.Info("ACT grantees listed")
+
+	// download act file with added grantees
+	grantees := []string{"02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e8",
+		"02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e9",
+		"02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12ee"}
+
+	for _, str := range grantees {
+		gPublisher, _ := swarm.ParseHexAddress(str)
+		history := file.HistroryAddress()
+		gSize, gHash, gPErr := upClient.DownloadActFile(ctx, file.Address(), &api.DownloadOptions{Act: &act, ActPublicKey: &gPublisher, ActHistoryAddress: &history, ActTimestamp: &timestamp})
```

Comment:
- Doesn't the file have only 1 publisher (the original uploader)?
- Shouldn't this be testing that every grantee can download the file?  i.e `granteeClient.DownloadActFile`
- Test that revoked clients cannot download the file is missing

Reply:
Yes, the file have only 1 publisher. Fixed
https://github.com/Solar-Punk-Ltd/beekeeper/blob/feat/act/pkg/check/act/act.go#L186